content/300-guides/100-performance-and-optimization/150-connection-management/index.mdx

```diff
@@ -0,0 +1,286 @@
+---
+title: Connection management
+metaTitle: Connection management (Guide)
+metaDescription: 'How to approach connection management with Prisma Client in serverless and PaaS applications.'
+tocDepth: 2
+---
+
+<TopBlock>
+
+Databases can handle a limited number of concurrent connections. Each connection requires RAM, which means that simply increasing the connection limit without scaling the database:
+
+* ✔ might allow more processes to connect _but_
+* ✘ significantly affects database **performance**
+
+The way your application **manages connections** also impacts performance. This guide describes how to approach connection management in [serverless environments](#serverless-environments-faas) and [long-running processes](#long-running-processes-paas).
+
+</TopBlock>
+
+## Long-running processes (PaaS)
+
+Examples of long-running processes include Node.js applications hosted on a PaaS service like Heroku. Use the following checklist as a guide to connection management in PaaS environments:
+
+- ☐ Tune the [connection limit](#recommended-connection-limit)
+- ☐ Make sure you have [**one** global instance of `PrismaClient`](#prismaclient-in-long-running-applications)
+
+### Recommended connection limit
+
+The recommended connection limit for long-running processes is the **default connection limit** (`num_physical_cpus * 2 + 1`) ÷ **number of application instances**.
+
+If you have **one** application instances:
+
+* The default connection limit applies by default (`num_physical_cpus * 2 + 1`) - you do not need to set the `connection_limit` parameter.
+* You can optionally [tune the pool size](#optimizing-the-connection-pool).
+
+If you have **multiple** application instances:
+
+* You must **manually** [set the `connection_limit` parameter](../../../concepts/components/prisma-client/working-with-prismaclient/connection-pool#setting-the-connection-limit) <span class="concept"></span>. For example, if your calculated connection limit is _10_ and you have _2_ instances of your app, the connection limit should be **no more than _5_**.
+* You can optionally [tune the pool size](#optimizing-the-connection-pool).
+
+### <inlinecode>PrismaClient</inlinecode> in long-running applications
+
+In **long-running** applications, we recommend that you:
+
+* ✔ Create **one** instance of `PrismaClient` and re-use it across your application
+* ✔ Assign `PrismaClient` to a global variable _in dev environments only_ to [prevent hot reloading from creating new instances](#prevent-hot-reloading-from-creating-new-instances-of-prismaclient)
+
+To re-use a single instance, create a module that exports a `PrismaClient` object:
+
+```ts file=client.ts
+import { PrismaClient } from "@prisma/client"
+
+let prisma = new PrismaClient() 
+
+export default prisma
+```
+
+The object is [cached](https://nodejs.org/api/modules.html#modules_caching) the first time the module is imported. Subsequent requests return the cached object rather than creating a new `PrismaClient`:
```

Comment:
Is this really required? In an express app the import would only happen once anyway so this makes it to complicated. Mayybe mention that option as well?